content/en/blog/2021/wasm-progress/index.md

```diff
@@ -0,0 +1,68 @@
+* Provides a guide for Wasm extension development, testing, and release. The guide is based on the same build tool chains and test frameworks that are used, maintained and tested by the Istio extensibility team.
+
+Currently the existing guide covers [WebAssembly extension development](https://github.com/istio-ecosystem/wasm-extensions/blob/master/doc/write-a-wasm-extension-with-cpp.md)
+and [unit testing](https://github.com/istio-ecosystem/wasm-extensions/blob/master/doc/write-cpp-unit-test.md) with C++,
+as well as [integration testing](https://github.com/istio-ecosystem/wasm-extensions/blob/master/doc/write-integration-test.md) with a GoLang test framework,
+which simulates real runtime by running a Wasm module with the Istio proxy binary.
+In the future, several more canonical extensions will also be added such as open policy agent, and header manipulation based on JWT token.
+
+## Wasm Module Distribution via Istio Agent
+
+Prior to Istio 1.9, [Envoy remote data source](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto#config-core-v3-remotedatasource) were needed to distribute remote Wasm modules to the proxy.
+An example configuration could be found [here](https://gist.github.com/bianpengyuan/8377898190e8052ffa36e88a16911910),
+where two `EnvoyFilter` resources are defined: one is to add a remote fetch Envoy cluster, and the other one is to inject a Wasm filter into HTTP filter chain.
+However this method has a drawback: if remote fetch fails either due to bad configuration or transient error, Envoy will be stuck with the bad configuration.
+If a Wasm extension is configured as [fail closed](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/wasm/v3/wasm.proto#extensions-wasm-v3-pluginconfig), a bad remote fetch will stop Envoy from serving.
+To fix this issue, [a fundamental change](https://github.com/envoyproxy/envoy/issues/9447) is needed to Envoy xDS protocol to make it allow asynchronous xDS response.
+
+Istio 1.9 provides a reliable distribution mechanism out of the box by leveraging the XDS proxy inside istio-agent and Envoy [Extension Configuration Discovery Server](https://www.envoyproxy.io/docs/envoy/latest/configuration/overview/extension) (ECDS).
+The following diagram shows the flow.
+Istio-agent intercepts the extension config resource update from istiod, reads the remote fetch hint from it, downloads the Wasm module, and rewrites the ECDS configuration with the path of the downloaded Wasm module.
+If download fails, istio-agent will reject the ECDS update and prevent bad configuration reaching Envoy. For more detail, please go through [this guide](/docs/ops/configuration/extensibility/wasm-module-distribution/).
+
+{{< image width="75%"
+    link="./architecture-istio-agent-downloading-wasm-module.svg"
+    alt="Remote Wasm module fetch flow"
+    caption="Remote Wasm module fetch flow"
+    >}}
+
+## Istio Wasm SIG and Future Work
```

Comment:
SIG = Working Group?

Reply:
In my understanding, Wasm SIG is like a subgroup of extensibility and telemetry WG. It has a very focused topic, which is Wasm based extension. @mandarjog we really need something to link here, so that people understands wasm sig better and join if they like.

Reply:
can we use https://tinyurl.com/istio-wasm-todo ?

Reply:
Yeah I guess that is the only available detail link. Added a link to it. Thanks!